python/pyiceberg/expressions/__init__.py

```diff
@@ -18,18 +18,37 @@
 from __future__ import annotations
 
 from abc import ABC, abstractmethod
-from dataclasses import dataclass
 from functools import reduce
-from typing import ClassVar, Generic, TypeVar
-
-from pyiceberg.expressions.literals import Literal
+from typing import (
+    Any,
+    Generic,
+    Iterable,
+    Set,
+    Type,
+    Union,
+)
+
+from pyiceberg.expressions.literals import Literal, literal
 from pyiceberg.files import StructProtocol
 from pyiceberg.schema import Accessor, Schema
+from pyiceberg.typedef import L
```

Comment:
Renaming the generic `T` to `L` requires a ton of changes and I'm not sure that it is worth doing. The only benefit is that we are getting a type bound on `T`, right?

Reply:
That is true, but I like the fact that we have all the bound types in one place.